mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/quota/CassandraQuotaLimitDao.java

```diff
@@ -0,0 +1,209 @@
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.cassandra.quota;
+
+import static com.datastax.oss.driver.api.querybuilder.QueryBuilder.bindMarker;
+import static com.datastax.oss.driver.api.querybuilder.QueryBuilder.deleteFrom;
+import static com.datastax.oss.driver.api.querybuilder.QueryBuilder.insertInto;
+import static com.datastax.oss.driver.api.querybuilder.QueryBuilder.selectFrom;
+import static com.datastax.oss.driver.api.querybuilder.relation.Relation.column;
+import static org.apache.james.mailbox.cassandra.table.CassandraQuotaLimitTable.IDENTIFIER;
+import static org.apache.james.mailbox.cassandra.table.CassandraQuotaLimitTable.QUOTA_COMPONENT;
+import static org.apache.james.mailbox.cassandra.table.CassandraQuotaLimitTable.QUOTA_LIMIT;
+import static org.apache.james.mailbox.cassandra.table.CassandraQuotaLimitTable.QUOTA_SCOPE;
+import static org.apache.james.mailbox.cassandra.table.CassandraQuotaLimitTable.QUOTA_TYPE;
+import static org.apache.james.mailbox.cassandra.table.CassandraQuotaLimitTable.TABLE_NAME;
+
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+import org.apache.james.backends.cassandra.utils.CassandraAsyncExecutor;
+import org.apache.james.core.quota.QuotaComponent;
+import org.apache.james.core.quota.QuotaLimit;
+import org.apache.james.core.quota.QuotaScope;
+import org.apache.james.core.quota.QuotaType;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.cql.PreparedStatement;
+import com.datastax.oss.driver.api.core.cql.Row;
+import com.datastax.oss.driver.api.querybuilder.delete.Delete;
+import com.datastax.oss.driver.api.querybuilder.insert.Insert;
+import com.datastax.oss.driver.api.querybuilder.select.Select;
+import com.google.common.base.MoreObjects;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class CassandraQuotaLimitDao {
+
+    public static class QuotaLimitKey {
```

Comment:
Just use QuotaKey + Scope?

Reply:
sr I do not get your idea

Reply:
QuotaLimitKey = QuotaKey + Scope so that we do not need to duplicate some fiels.

But maybe I am too picky. What is currently implemented is fine to me.